mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Defines utility functions for classical shadows protocol."""
+from typing import Tuple, List, Any, Union
+import mitiq
+import cirq
+import numpy as np
+from cirq.ops.pauli_string import PauliString
+from numpy.typing import NDArray
+import re
+
+
+def n_measurements_tomography_bound(epsilon: float, num_qubits: int) -> int:
+    """
+    This function returns the minimum number of classical shadows required
+    for state reconstruction for achieving the desired accuracy.
+
+    Args:
+        epsilon: The error on the estimator.
+        num_qubits: The number of qubits in the system.
+
+    Returns:
+        An integer that gives the number of snapshots required to satisfy the
+        shadow bound.
+    """
+    return int(34 * (4**num_qubits) * epsilon ** (-2))
+
+
+def local_clifford_shadow_norm(opt: PauliString[Any]) -> float:
+    """
+    Calculate shadow norm of an operator with random unitary sampled from local
+    Clifford group.
+    Args:
+        opt: a self-adjoint operator
+    Returns:
+        Shadow norm when unitary ensemble is local Clifford group.
+    """
+    norm = (
+        np.linalg.norm(
+            cirq.unitary(opt)
+            - np.trace(cirq.unitary(opt))
+            / 2 ** int(np.log2(cirq.unitary(opt).shape[0])),
+            ord=np.inf,
+        )
+        ** 2
+    )
```

Comment:
defining `unitary = cirq.unitary(opt)` will speed this up a bit so it doesn't have to be computed 3 times. If speed is not an issue (even with large `opt`s, feel free to ignore), but it would [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) the code up a bit.

Reply:
I have rewritten this to mitiq.PauliString, and this line becomes 
```
opt = obs.matrix()
    norm = (
        np.linalg.norm(
            opt - np.trace(opt) / 2 ** int(np.log2(opt.shape[0])),
            ord=np.inf,
        )
        ** 2
    )
```